src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigRunners.cs

```diff
@@ -54,5 +74,57 @@ public static ISettings GetSettingsFromDirectory(string directory)
                 configFileName: null,
                 machineWideSettings: new XPlatMachineWideSetting());
         }
+
+        /// <summary>
+        /// Creates a settings object utilizing a NuGet configuration file path.
+        /// </summary>
+        /// <param name="filePath"></param>
+        /// <returns></returns>
+        public static ISettings GetSettingsFromFile(string filePath)
+        {
+            var directory = Path.GetDirectoryName(filePath);
+            return NuGet.Configuration.Settings.LoadDefaultSettings(
+                directory,
+                configFileName: filePath,
+                machineWideSettings: new XPlatMachineWideSetting());
+        }
+
+        /// <summary>
+        /// Throws an exception if any of the config runner arguments are null.
+        /// </summary>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static void ValidateArguments<TArgs>(TArgs args, Func<ILogger> logger)
+        {
+            if (args == null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+        }
+
+        /// <summary>
+        /// Throws an exception if the value passed in is not a valid config key.
+        /// </summary>
+        /// <exception cref="CommandException"></exception>
+        public static void ValidateConfigKey(string configKey)
+        {
+            if (!configKey.Equals(ConfigurationConstants.DependencyVersion, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.GlobalPackagesFolder, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.RepositoryPath, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.DefaultPushSource, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.HostKey, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.UserKey, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.PasswordKey, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.NoProxy, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.MaxHttpRequestsPerSource, StringComparison.OrdinalIgnoreCase)
+                && !configKey.Equals(ConfigurationConstants.SignatureValidationMode, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new CommandException(string.Format(CultureInfo.CurrentCulture, Strings.Error_ConfigSetInvalidKey, configKey));
```

Comment:
Maybe this if statements its too large. I was looking at enums and maybe you can use TryParse from https://learn.microsoft.com/en-us/dotnet/api/system.enum.tryparse?view=net-7.0.

You can pass a boolean as parameter to do the operation  case-insensitive